gobblin-runtime/src/main/java/org/apache/gobblin/runtime/spec_store/FSSpecStore.java

```diff
@@ -98,17 +98,39 @@ public FSSpecStore(Config sysConfig, SpecSerDe specSerDe, Optional<Logger> log)
     }
   }
 
+  public static String getFlowGroup(Path specUri){
+    return specUri.getParent().getName();
+  }
+
+  public static String getFlowName(Path specUri){
+    return Files.getNameWithoutExtension(specUri.getName());
+  }
+
   @Override
   public boolean exists(URI specUri) throws IOException {
     Preconditions.checkArgument(null != specUri, "Spec URI should not be null");
-
+    Path flowPath = new Path(specUri.getPath());
+    String flowGroup = getFlowGroup(flowPath);
+    String flowName = getFlowName(flowPath);
     FileStatus[] fileStatuses = fs.listStatus(this.fsSpecStoreDirPath);
+
     for (FileStatus fileStatus : fileStatuses) {
-      if (StringUtils.startsWith(fileStatus.getPath().getName(), specUri.toString())) {
+      if (fileStatus.isDirectory() && fileStatus.getPath().getName().equals(flowGroup)) {
+        return exists(fileStatus.getPath(), flowName);
+      } else if(!fileStatus.isDirectory() && fileStatus.getPath().getName().startsWith(flowName)) {
```

Comment:
There should always be a flow group, so the else case should never apply.

Reply:
I started with that only; but realized that TopologyCatalog also uses the same methods.
https://github.com/apache/incubator-gobblin/blob/master/gobblin-runtime/src/main/java/org/apache/gobblin/runtime/spec_catalog/TopologyCatalog.java#L257
and topologies do not have group directory.
Maybe we should implement separate method for TopologyCatalog.